lib/web_ui/lib/src/engine/canvaskit/text_fragmenter.dart

```diff
@@ -0,0 +1,91 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:typed_data';
+
+import '../dom.dart';
+import '../text/line_breaker.dart';
+
+abstract class CkTextFragmenter {
+  const CkTextFragmenter(this.text);
+
+  /// The text to be fragmented.
+  final String text;
+
+  /// Performs the fragmenting of [text] and returns a [Uint32List].
+  Uint32List fragment();
+}
+
+/// The granularity at which to segment text.
+///
+/// To find all supported granularities, see:
+/// - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Segmenter/Segmenter
+enum IntlSegmenterGranularity {
+  grapheme,
+  word,
+}
+
+class CkIntlFragmenter extends CkTextFragmenter {
+  CkIntlFragmenter(super.text, this.granularity)
+      : assert(domIntl.Segmenter != null);
+
+  final IntlSegmenterGranularity granularity;
+
+  String get _granularityString {
```

Comment:
This can become a `final String _granularityString` to avoid putting the switch block on every `fragment()`  invocation.

Reply:
Not applicable anymore.